### PR TITLE
tf.negative

### PR DESCRIPTION
**Pull request by yanndupis**
Hey @mortendahl,

This PR implements the [tf.negative](https://www.tensorflow.org/api_docs/python/tf/math/negative) operation (`tf.negative([-1, 0, 1]) == [1, 0, -1]`). I named it `neg` so it doesn't get mixed with the negative operation in the securenn protocol (`negative([-1, 0, 1]) == [1, 0, 0]`).  

This operation appeared in one of the convolution network when I turned off in the batchnorm layer the parameters scale and offset. 

Thanks!

### CODE REVIEW

**Comment by mortendahl** [comment]
hi @yanndupis, good work! I suggest we rename the existing method in SecureNN, maybe to `is_negative`, in order to stick with the TF API; anything else will only cause more confusions down the line.

File: tf_encrypted/protocol/pond.py [hunk 1]
```diff
@@ -3526,6 +3556,77 @@ def _reshape_masked(
...
+        with tf.device(prot.server_0.device_name):
+            x0_neg = x0.neg()
+
+        with tf.device(prot.server_1.device_name):
+            x1_neg = x1.neg()
+
+        return PondPrivateTensor(prot, x0_neg, x1_neg, x.is_scaled)
+
+
+def _neg_masked(
```
Comment (mortendahl) [inline tf_encrypted/protocol/pond.py]: this works out but just a small note of caution that this is not always straight-forward and could lead to privacy issues 👍

**Comment by yanndupis** [comment]
Hey @mortendahl ,

Thank you very much for your review. 

I have realized that `negative` was already implemented on the backing tensor. 

What do you think is better in `int100.py`: 
```
def negative(self):
            # TODO[Morten] there's probably a more efficient way
            return FACTORY.zero() - self
```
 or 

```
def neg(self):
            backing = [tf.math.negative(xi) for xi in self.backing]
            return DenseTensor(backing)
```

and in `native.py`:

```
def negative(self):
            value = tf.negative(self.value)
            if EXPLICIT_MODULUS is not None:
                value %= EXPLICIT_MODULUS
            return DenseTensor(value)
``` 

or 

```
def neg(self):
            return DenseTensor(tf.negative(self.value))
```

Thank you!

**Comment by mortendahl** [comment]
Hi @yanndupis, my suggestion:

- For int100:

```python
def neg(self):
    backing = [tf.negative(xi) % mi for xi, mi in zip(self.backing, MODULI)]
    return DenseTensor(backing)
```

- For native:

```python
def negative(self):
    value = tf.negative(self.value)
    if EXPLICIT_MODULUS is not None:
        value %= EXPLICIT_MODULUS
    return DenseTensor(value)
```

**Comment by yanndupis** [comment]
Hey @mortendahl , thanks for your review! Your comments should be addressed with this pr. thanks.
